x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/get_agent_marks.ts

```diff
@@ -8,20 +8,23 @@ import { sortBy } from 'lodash';
 import { Transaction } from '../../../../../../typings/es_schemas/ui/Transaction';
 
 export interface AgentMark {
+  docType: 'agentMark';
   name: string;
-  us: number;
+  offset: number;
 }
 
-export function getAgentMarks(transaction: Transaction): AgentMark[] {
-  if (!(transaction.transaction.marks && transaction.transaction.marks.agent)) {
+export function getAgentMarks(transaction?: Transaction): AgentMark[] {
+  const agent = transaction?.transaction.marks?.agent;
+  if (!agent) {
     return [];
   }
 
   return sortBy(
-    Object.entries(transaction.transaction.marks.agent).map(([name, ms]) => ({
+    Object.entries(agent).map(([name, ms]) => ({
+      docType: 'agentMark',
```

Comment:
I think `markType` is better. This also more clearly separates it from the `docType` in waterfall items.
```suggestion
      markType: 'agentMark',
```